web/models/summary_helper.ex

```diff
@@ -71,4 +71,18 @@ defmodule CoursePlanner.SummaryHelper do
        |> List.first
   end
   def get_next_class(_offered_courses), do: nil
+
+  def get_next_task(user, time \\ Timex.now())
+  def get_next_task(%{id: user_id, role: "Volunteer"}, time) do
+    all_tasks =
+      Repo.all(from t in Task,
+        join: v in assoc(t, :volunteers),
+        preload: [volunteers: v],
+        where: v.id == ^user_id and t.start_time >= ^time)
+
+    all_tasks
+    |> Enum.sort(&(&1.start_time <= &2.start_time))
+    |> List.first
```

Comment:
you can do the sorting and first on the query also, please check https://hexdocs.pm/ecto/Ecto.Query.html#first/2 for reference